app/controllers/katello/api/v2/subscriptions_controller.rb

```diff
@@ -22,20 +19,24 @@ class Api::V2::SubscriptionsController < Api::V2::ApiController
     api :GET, "/organizations/:organization_id/subscriptions", N_("List organization subscriptions")
     api :GET, "/systems/:system_id/subscriptions", N_("List a system's subscriptions"), :deprecated => true
     api :GET, "/activation_keys/:activation_key_id/subscriptions", N_("List an activation key's subscriptions")
+    api :GET, "/subscriptions"
+    param_group :search, Api::V2::ApiController
     param :organization_id, :number, :desc => N_("Organization ID"), :required => true
     param :system_id, String, :desc => N_("UUID of the system"), :required => false
     param :activation_key_id, String, :desc => N_("Activation key ID"), :required => false
-    param_group :search, Api::V2::ApiController
     def index
-      subscriptions = if @system
-                        index_system
-                      elsif @activation_key
-                        index_activation_key
-                      else
-                        index_organization
-                      end
+      respond(:collection => scoped_search(index_relation.uniq, :cp_id, :asc, :resource_class => Pool))
+    end
 
-      respond_for_index(:collection => subscriptions)
+    def index_relation
+      collection = Pool.scoped
```

Comment:
I was surprised to see the subscriptions controller doing everything from the pool perspective. Grant you, I probably don't know this area as well as you do by now, so I am assuming that even though we are presenting "subscriptions" to the user, we are in reality presenting "Candlepin pools". Is that a fair assessment?